src/nvim/msgpack_rpc/channel.c

```diff
@@ -343,39 +348,41 @@ static void handle_request(Channel *channel, msgpack_object *request)
 
     if (is_get_mode && !input_blocking()) {
       // Defer the event to a special queue used by os/input.c. #6247
-      multiqueue_put(ch_before_blocking_events, on_request_event, 1, evdata);
+      multiqueue_put(ch_before_blocking_events, response_event, 1, evdata);
     } else {
       // Invoke immediately.
-      on_request_event((void **)&evdata);
+      response_event((void **)&evdata);
     }
   } else {
-    multiqueue_put(channel->events, on_request_event, 1, evdata);
+    multiqueue_put(channel->events, response_event, 1, evdata);
     DLOG("RPC: scheduled %.*s", method->via.bin.size, method->via.bin.ptr);
   }
 }
 
-static void on_request_event(void **argv)
+/// Responds to a message, depending on the type:
+///   - Request: writes the response.
+///   - Notification: does nothing.
+static void response_event(void **argv)
```

Comment:
This name and description is a bit misleading, the main thing this function does is handling the incoming request (at a time where it is safe to do so), and as result of that it might send a response. If this function "does nothing" on a notification, there would be no point in sending notifications.